lib/_tls_common.js

```diff
@@ -249,3 +250,16 @@ exports.translatePeerCertificate = function translatePeerCertificate(c) {
   }
   return c;
 };
+
+exports.derCertToPemCert = function derCertToPemCert(cert) {
+  const chars = Buffer.from(cert).toString('base64').split('');
+  let formattedCert = '-----BEGIN CERTIFICATE-----';
+  for (var i = 0; i < chars.length; i++) {
+    if (i % 64 === 0) {
+      formattedCert += '\n';
+    }
+    formattedCert += chars[i];
+  }
```

Comment:
This is a horribly inefficient way of building a string.  Try this:
```js
const slices = ['-----BEGIN CERTIFICATE-----'];
for (var i = 0; i < chars.length; i += 64)
  slices.push(chars.slice(i, i + 64));
slices.push('-----END CERTIFICATE-----');
return slices.join('\n');
```

Reply:
You could also do:

```js
return '-----BEGIN CERTIFICATE-----\n' +
       cert.toString('base64').replace(/.{64}/g, '$&\n') +
       (cert.length % 64 ? '\n' : '') +
       '-----END CERTIFICATE-----\n';
```